### ISSUE TITLE
IDE0044 (make field readonly) not fired when field is dereferenced within class

### ISSUE THREAD

**Issue by drewnoakes** [dotnet/roslyn#29746]
**Version Used**: VS Enterprise 15.8.3

**Steps to Reproduce**:

```c#
public class Foo
{
    private string _s;
    public Foo(string s) => _s = s;
    public string Bar()
    {
        //return _s;           // IDE0044 triggers on field
        //return _s.ToUpper(); // IDE0044 does not trigger
    }
}
```

**Expected Behavior**:

Diagnostic IDE0044 should fire.

**Actual Behavior**:

No diagnostic.
